server/middleware/middleware.go

```diff
@@ -45,6 +45,10 @@ func (rh *responseHijacker) Write(b []byte) (int, error) {
 	return rh.ResponseWriter.Write(b)
 }
 
+func (rh *responseHijacker) CloseNotify() <-chan bool {
+	return rh.ResponseWriter.(http.CloseNotifier).CloseNotify()
```

Comment:
It's not clear if gohan response writer must implement CloseNotifier interface or not. In https://github.com/cloudwan/gohan/pull/377/files#diff-457a66ed8b4911098a4e6bc1f0be58f7R178 you check if type assertion is safe, here the code would panic if `ResponseWriter` is not `CloseNotifier`. WDYT?